cajita/unit_test/tstSparseDimPartitioner.hpp

```diff
@@ -0,0 +1,293 @@
+#include <Cajita_SparseDimPartitioner.hpp>
+#include <Cajita_SparseIndexSpace.hpp>
+#include <Kokkos_Core.hpp>
+
+#include <gtest/gtest.h>
+
+#include <mpi.h>
+
+using namespace Cajita;
+
+namespace Test
+{
+void uniform_distribution_automatic_rank()
```

Comment:
Can you add some more detail about the problem setup? Does the partitioning actually change?